src/main/scala/LumberJack.scala

```diff
@@ -0,0 +1,118 @@
+import java.util
+
+import LumberJack.currentPiles
+
+import scala.collection.mutable.ArrayBuffer
+import scala.io.StdIn.readLine
+
+object LumberJack extends App
+{
+
+  //val grid = takeInput().toInt
+  //val logAmount = takeInput().toInt
+  //var currentPiles: Array[Int][Int] = { (0, 0 , 1) (0, 1 , 1) (0 , 0 , 1) }
+  var currentPiles = Array.ofDim[Int](3, 3)
+  var pileAmount: Array[Int] = Array(1, 2 ,3, 4 , 5 , 6 ,7 ,8 ,9 )
+  currentPiles = genrateDisplay(currentPiles, pileAmount)
+  currentPiles foreach{case a =>  a foreach{b => print(b.toString + " ")}; print('\n') }
+  println("")
+  var logAmount: Int = 7
+
+  pileAmount = adjustLogPile(pileAmount, genrateDisplay(currentPiles, pileAmount) , logAmount)
+  currentPiles = genrateDisplay(currentPiles, pileAmount)
+  pileAmount foreach( a => println(a))
+  currentPiles foreach{case a =>  a foreach{b => print(b.toString + " ")}; print('\n') }
+
+
+  def adjustLogPile(pileAmount: Array[Int] , currentPiles: Array[Array[Int]] , logAmount: Int): Array[Int] =
+  {
+
+      var cLogAmount = logAmount
+      while (cLogAmount > 0)
+      {
+        for (i <- 0 to pileAmount.size - 1)
+        {
+          var smallestNum = findsmallestNum(currentPiles)
+          if (pileAmount(i) == smallestNum)
+          {
+            pileAmount(i) = pileAmount(i) + 1
+            cLogAmount = cLogAmount - 1
+            println("the amount of logs is " + cLogAmount)
+            //check the adjustment for each iteration
+            println("numbers found ")
+            println(logAmount)
+            pileAmount foreach( a => print(a))
+          }
+        }
+      }
+      pileAmount
+    }
+
+
+  def findsmallestNum(currentPiles: Array[Array[Int]]): Int =
+  {
+    var smallestNum = currentPiles(0)(0)
+    for (k <- 0 to currentPiles.size - 1 )
```

Comment:
remember that you can have one for loop and give it two ranges as well as use the guards, making your two for loops and if statement more clean